Firmware/first_lay_cal.cpp

```diff
@@ -212,3 +213,33 @@ void lay1cal_square(char *cmd_buffer, uint8_t i)
     sprintf_P(cmd_buffer, fmt2, (35 - (i + 1)*width * 2), extr_short_segment);
     enquecommand(cmd_buffer);
 }
+
+
+//! @brief Set the flow in order to the installed nozzle
+//!
+//! This function must be called before to start to print in order to 
+//! enqueue the correct flow depending the nozzle size stored on the 
+//! eeprom
+//!
+void lay1cal_set_flow()
+{    
+    ClNozzleDiameter nDiameter=(ClNozzleDiameter)eeprom_read_byte((uint8_t*)EEPROM_NOZZLE_DIAMETER);
+
+    switch(nDiameter){
+    case ClNozzleDiameter::_Diameter_250:
+        enquecommand_P(PSTR("M221 S62"));
+        break;
+    case ClNozzleDiameter::_Diameter_400:
```

Comment:
This could be:

```
case ClNozzleDiameter::_Diameter_Undef:
case ClNozzleDiameter::_Diameter_400:
```

so that we fold the undefined case with the .4 nozzle.

Reply:
I don't know why, but the solution that you propose is expensive in bytes, the use of program storage space is `251058` 

```
Start to build Prusa Firmware ...
Using variant 1_75mm_MK3S-EINSy10a-E3Dv6full
Sketch uses **251058** bytes (98%) of program storage space. Maximum is 253952 bytes.
Global variables use 6103 bytes of dynamic memory.
```
But with the current solution, the bytes used are `251018`.

```
Start to build Prusa Firmware ...
Using variant 1_75mm_MK3S-EINSy10a-E3Dv6full
Sketch uses **251018** bytes (98%) of program storage space. Maximum is 253952 bytes.
Global variables use 6103 bytes of dynamic memory.
```
I suppose that the compiler identifies the same static string and just links to that string and because of that consume fewer bytes because the string is not really duplicated, anyway, the best approach is the current one, what do you think?
Maybe I'm wrong, I need to investigate a little bit about that behavior.

Reply:
I just tried with the following:

```
    switch(nDiameter){
    case ClNozzleDiameter::_Diameter_250:
        enquecommand_P(PSTR("M221 S62"));
        break;
    case ClNozzleDiameter::_Diameter_Undef:
    case ClNozzleDiameter::_Diameter_400:
        enquecommand_P(PSTR("M221 S100"));
        break;
    case ClNozzleDiameter::_Diameter_600:
        enquecommand_P(PSTR("M221 S150"));
        break;
    case ClNozzleDiameter::_Diameter_800:
        enquecommand_P(PSTR("M221 S200"));
        break;
    }
```

and it shrinks by ~14 bytes on my end

Reply:
You were right, I have pushed the changes just to compare the Travis build logs. 

The las build with your suggested code:

```
Sketch uses 251412 bytes (98%) of program storage space. Maximum is 253952 bytes.

Global variables use 6108 bytes of dynamic memory.
```

The previous build from 10 days ago:

```
Sketch uses 251482 bytes (99%) of program storage space. Maximum is 253952 bytes.

Global variables use 6106 bytes of dynamic memory.
```

So the code consumes fewer bytes now.